File: Backend/app/models/job_application.py
from sqlalchemy import Column, String, Text, Boolean, Date, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from datetime import datetime, timezone
import uuid
from app.database import Base

class JobApplication(Base):
    __tablename__ = "job_applications"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"))
    company = Column(String(255), nullable=False)
    role_title = Column(String(255), nullable=False)
    status = Column(String(50), default="WISHLIST")
    job_url = Column(Text)
    notes = Column(Text)
    location = Column(String(255))
    is_remote = Column(Boolean, default=False)
    applied_date = Column(Date)
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
    updated_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))

File: Frontend/applyflow/lib/models/job_application.dart
class JobApplication {
  final String id;
  final String company;
  final String roleTitle;
  final String status;
  final String? jobUrl;
  final String? notes;
  final String? location;
  final DateTime? appliedDate;
  final DateTime createdAt;

  JobApplication({
    required this.id,
    required this.company,
    required this.roleTitle,
    required this.status,
    this.jobUrl,
    this.notes,
    this.location,
    this.appliedDate,
    required this.createdAt,
  });

  factory JobApplication.fromJson(Map<String, dynamic> json) {
    return JobApplication(
      id: json['id'] ?? '',
      company: json['company'] ?? '',
      roleTitle: json['role_title'] ?? '',
      status: json['status'] ?? 'WISHLIST',
      jobUrl: json['job_url'],
      notes: json['notes'],
      location: json['location'],
      appliedDate: json['applied_date'] != null
          ? DateTime.tryParse(json['applied_date'])
          : null,
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at']) ?? DateTime.now()
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'company': company,
      'role_title': roleTitle,
      'status': status,
      'job_url': jobUrl?.isEmpty == true ? null : jobUrl,
      'notes': notes?.isEmpty == true ? null : notes,
      'location': location?.isEmpty == true ? null : location,
      'applied_date': appliedDate?.toIso8601String().split('T').first,
    };
  }
}
